tests/test_dump_file.py

```diff
@@ -187,3 +187,39 @@ def test_copy_from(snap, tmp_path):
     assert numpy.all(read_snap.molecule == [0, 1, 2])
     assert read_snap.has_charge()
     assert numpy.allclose(read_snap.charge, [0, 1, -1])
+
+
+@pytest.mark.parametrize(
+    "schema",
+    ["x", "x y", "vx", "vx vy", "ix", "ix iy"],
+    ids=[
+        "position-1d",
+        "position-2d",
+        "velocity-1d",
+        "velocity-2d",
+        "image-1d",
+        "image-2d",
+    ],
+)
+def test_faulty_dump_schema(snap, tmp_path, schema):
+    # test schema
+    schema_zeros = [0] * len(schema.split())
+    print(schema_zeros)
+    f = open(tmp_path/"atoms.lammpstrj", "w")
+    f.write(
+        "ITEM: TIMESTEP\n"
+        "0 \n"
+        "ITEM: NUMBER OF ATOMS \n"
+        "1 \n"
+        "ITEM: BOX BOUNDS pp pp pp\n"
+        "-7.0 7.0 \n"
+        "-7.0 7.0 \n"
+        "-7.0 7.0 \n"
+        f"ITEM: ATOMS id type {schema}\n"
+        f"1 1 {" ".join(str(i) for i in schema_zeros)} \n"
+    )
+    f.close()
+    filename = tmp_path/"atoms.lammpstrj"
+
+    with pytest.raises(IOError):
+        lammpsio.DumpFile(filename)
```

Comment:
The OP reported needing this loop over snapshots in the trajectory to trigger the error. I think that's right because constructing the class doesn't open the file immediately.
```suggestion

    traj = lammpsio.DumpFile(filename)
    with pytest.raises(IOError):
        for snap in traj:
            pass
```